zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/compensation/CompensationEventExecutionTest.java

```diff
@@ -1316,6 +1316,44 @@ public void shouldPropagateVariablesOfCompensationHandler() {
             tuple(processInstanceKey, "global", "2"));
   }
 
+  @Test
+  public void shouldTriggerCompensationForMultiInstanceActivityOnlyOnce() {
+    final var process =
+        createModelFromClasspathResource("/compensation/compensation-multi-instance-activity.bpmn");
+    ENGINE.deployment().withXmlResource(process).deploy();
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withType(SERVICE_TASK_TYPE_COMPENSABLE_ACTIVITY)
+        .limit(3)
+        .forEach(job -> ENGINE.job().withKey(job.getKey()).complete());
+
+    ENGINE
+        .job()
+        .ofInstance(processInstanceKey)
+        .withType(SERVICE_TASK_TYPE_COMPENSATION_HANDLER)
+        .complete();
+
+    // then
+    assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .limit(6))
+        .extracting(Record::getIntent, r -> r.getValue().getCompensationHandlerId())
+        .containsSubsequence(
+            tuple(CompensationSubscriptionIntent.TRIGGERED, "CompensationHandler"),
+            tuple(CompensationSubscriptionIntent.COMPLETED, "CompensationHandler"));
```

Comment:
:x: The expected number of compensation subscriptions is not correct. 

For a multi-instance activity with a compensation boundary event, we should create only one subscription. The subscription should belong to the multi-instance body instead of the inner activity.  

Currently, we create two subscriptions: one for the multi-instance body and one for the inner activity.